src/auto_pickup.cpp

```diff
@@ -209,70 +212,70 @@ void auto_pickup::show( const std::string &custom_name, bool is_autopickup )
         const std::string action = ctxt.handle_input();
 
         if (action == "NEXT_TAB") {
-            iCurrentPage++;
-            if (iCurrentPage > 2) {
-                iCurrentPage = 1;
-                iCurrentLine = 0;
+            iTab++;
+            if (iTab > CHARACTER_TAB) {
+                iTab = GLOBAL_TAB;
+                iLine = 0;
             }
         } else if (action == "PREV_TAB") {
-            iCurrentPage--;
-            if (iCurrentPage < 1) {
-                iCurrentPage = 2;
-                iCurrentLine = 0;
+            iTab--;
+            if (iTab < GLOBAL_TAB) {
```

Comment:
The usage of specific enum values is a bit brittle. If somebody changes the order or even just adds a new value before `MAX`, the code will break. It's probably better to compare the `MAX` value (which should always be the last value of the enumeration) and 0 (which is always the first value).

The current logic is also quite difficult to understand:
```C++
if (iTab < GLOBAL_TAB) { // arbitrary value
    iTab = CHARACTER_TAB; // arbitrary value
}
```
vs.
```C++
if (iTab < 0) {
    iTab = MAX_TAB - 1;
}
```

Similar for the other check above.